packages/eslint-plugin/tests/rules/no-unnecessary-condition.test.ts

```diff
@@ -2766,5 +2794,111 @@ isString('fa' + 'lafel');
       '((string & { __brandA: string }) | (number & { __brandB: string })) & ("foo" | 123)',
       'alwaysTruthy',
     ),
+    {
+      code: `
+type A = {
+  [name in Lowercase<string>]?: {
+    [name in Lowercase<string>]: {
+      a: 1;
+    };
+  };
+};
+
+declare const a: A;
+
+a.a?.a?.a;
+      `,
+      errors: [
+        {
+          column: 7,
+          endColumn: 9,
+          endLine: 12,
+          line: 12,
+          messageId: 'neverOptionalChain',
+        },
+      ],
+      output: `
+type A = {
+  [name in Lowercase<string>]?: {
+    [name in Lowercase<string>]: {
+      a: 1;
+    };
+  };
+};
+
+declare const a: A;
+
+a.a?.a.a;
+      `,
+    },
+    {
+      code: `
+interface T {
+  [name: Lowercase<string>]: {
+    [name: Lowercase<string>]: {
+      [name: Lowercase<string>]: {
+        vale: 'value';
```

Comment:
Here and a couple other places:

```suggestion
        value: 'value';
```

Reply:
oops thanks! https://github.com/typescript-eslint/typescript-eslint/pull/10460/commits/4fd032f9075acf0ec648acd82dee1833045e30a0